src/components/common/RadioButtonGroup.tsx

```diff
@@ -0,0 +1,100 @@
+import {
+  ButtonGroup,
+  ButtonGroupProps,
+  ButtonProps,
+  Icon,
+  UseRadioGroupProps,
+  UseRadioProps,
+  useRadio,
+  useRadioGroup,
+} from "@chakra-ui/react"
+import Button from "./Button"
+
+interface RadioButtonOptionProps {
+  label?: string
+  icon?: React.ElementType
+  value: string
+}
+
+type RadioButtonGroupProps = {
+  options: RadioButtonOptionProps[]
+  chakraStyles?: ButtonGroupProps & ButtonProps
+} & UseRadioGroupProps
+
+const RadioButtonGroup = ({
+  options,
+  chakraStyles,
+  ...useRadioGroupProps
+}: RadioButtonGroupProps) => {
+  const { getRootProps, getRadioProps } = useRadioGroup(useRadioGroupProps)
+
+  const group = getRootProps()
+
+  const buttonBorderRadius =
+    chakraStyles?.borderRadius ??
+    (() => {
+      switch (chakraStyles?.size) {
+        case "xs":
+          return "md"
+        case "sm":
+        case "md":
+          return "lg"
+        case "lg":
+        case "xl":
+        case "2xl":
+          return "xl"
+        default:
+          return "md"
+      }
+    })()
+
+  return (
+    <ButtonGroup {...chakraStyles} {...group}>
+      {options.map((option) => {
+        const radioProps = getRadioProps({ value: option.value })
+        return (
+          <RadioButton
+            key={option.label}
+            {...option}
+            {...radioProps}
+            colorScheme={chakraStyles?.colorScheme}
+            borderRadius={buttonBorderRadius}
+          />
+        )
+      })}
+    </ButtonGroup>
+  )
+}
+
+type RadioButtonProps = {
+  colorScheme?: string
+  borderRadius?: any
```

Comment:
small impro: we could use `ButtonProps["colorScheme"]` & `ButtonProps["borderRadius"]` here, so we get nice code completion!